### PR TITLE
Use external binary to extract class name from stdin Java sources

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

**Comment by alexarchambault** [comment]
(Adding some more scaladoc right now…)

File: modules/build/src/main/scala/scala/build/internal/JavaParserProxy.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    *   either some class name (if one was found) or none (if none was found), or a
+    *   [[BuildException]]
+    */
+  def className(
+    archiveCache: ArchiveCache[Task],
+    content: Array[Byte],
+    logger: Logger
+  ): Either[BuildException, Option[String]] = either {
+
+    val versionOpt = Some("0.1.0")
```
Comment (Gedochao) [inline modules/build/src/main/scala/scala/build/internal/JavaParserProxy.scala]: wouldn't this always be picked over `Constants.javaClassNameVersion`, given how it's coded atm?
Why not just do `val version = Constants.javaClassNameVersion`?

**Comment by Gedochao** [comment]
I think it'd be useful to mention [java-class-name](https://github.com/scala-cli/java-class-name) among other libraries/forks we maintain ourselves in a separate `.md` doc, so that it's easier to look them up when contributing. 

(unless we already have a doc like that? in which case `java-class-name` should be mentioned there)

File: modules/build/src/main/scala/scala/build/internal/JavaParserProxy.scala (hunk 1 above)
Comment (alexarchambault) [reply modules/build/src/main/scala/scala/build/internal/JavaParserProxy.scala]: Should be fixed.

**Comment by alexarchambault** [comment]
> I think it'd be useful to mention [java-class-name](https://github.com/scala-cli/java-class-name) among other libraries/forks we maintain ourselves in a separate `.md` doc, so that it's easier to look them up when contributing. 

To me, that doc is https://github.com/scala-cli

**Comment by Gedochao** [comment]
> To me, that doc is https://github.com/scala-cli

Okay... but do we mention the `scala-cli` org anywhere in our docs?
You may be right that it's probably sufficient to make the list of repos there visible to our contributors

**Comment by alexarchambault** [comment]
> Okay... but do we mention the `scala-cli` org anywhere in our docs?

Just opened https://github.com/VirtusLab/scala-cli/pull/1075 for it.

**Review by Gedochao — APPROVED** [review]

**Pull request merged**